### PR TITLE
34 update readme

### PR DESCRIPTION
**Pull request by wrex**
Okay, this was a big, fun one.

Please review carefully:

1. Extensive edits to the README.md file. I deleted all the old screenshots and created a few new ones (we should add more). Note that I used sharp to reduce to resize and compress the raw PNGs considerably (old habit).

2. I updated to use `--USER-*` overrides (but I've not yet done anything for "internal" props).

3. I created two sample user override stylesheets. Check out rexs-overrides.css -- a royal theme! :smile: It's pretty neat, in stylus just check/uncheck the override file to use the override or ED-default colors.

Remaining work:

- ~~Separate external API vs. internal props as we've been discussing.~~
- ~~Update the userscript stylesheets to ensure they use the API as intended (I've not looked for this merge).~~

### CODE REVIEW

**Comment by wrex** [comment]
I changed naming convention we were discussing in Issue #59: 

`--ED-*` for external API that script writers should use, and `--EDI-*` for internal  properties that should only be used within the main file. All of the `--ED-*` variables can be overridden with a correspondingly named `--USER-*` variable.

**Comment by wrex** [comment]
Gotta say, I'm really digging my color scheme during reviews. The dashboard is a little more confusing (especially because I've not styled some of my scripts yet) but reviews and lessons look awesome. Very satisfying. :smile:

**Comment by wrex** [comment]
I updated the stroke-order stylesheet to use the brand color instead of `--ED-red` (now `--EDI-red`).

I left the ultimate timeline stylesheet alone, but we seem to have gone about things inside-out. 

I've tried to make it as easy as possible for script stylesheet-writers: they can just use the `--ED-*` props directly. They don't need to provide a fallback or define their own unique props unless they want to.

Rationale:

- Nobody will install these userscript stylesheets unless they have (or think they have) installed ED.
- The user scripts themselves already should have default styling to work with the default WK theme.
- If the main ED stylesheet _isn't_ loaded and someone loads the script stylesheet to override the CSS rules with ED colors, it shouldn't be a problem if things look weird (there won't be any ED coloring anywhere).
- Order of stylesheet loading still won't matter. The cascade just works with custom properties.

**Comment by wrex** [comment]
Ah! I figured out why self-study was bugging me. It wasn't using the ED correct/incorrect and text colors. Much better with the last commit.

**Comment by wrex** [comment]
Two more trivial commits (affecting `rexs-overrides.css` only). I warmed up the color of the lessons button in my overrides stylesheet slightly. That made me realize my text-light color was a little too dark.

It's hard to find an unsaturated, "de-emphasized" color in the open-props palette that still allows light text. I'll wait until the merge, but I'd like to figure out a way to override text colors over any given background color.

Here's how my overrides look, by the way:

dashboard

kanji

The heatmap and item inspector on the dashboard are using Breeze Dark theme colors. The slider in ultimate timeline still uses the system (OSX) blue slider color, but sliders are a notorious PITA to style. 

On the kanji page, the niai-visually-similar script picks up some styling from the `-item-info-injector` class which is pretty cool, but we'll need to add a stylesheet for it to override the R/K/V colors.

All in all, I'm quite happy with this branch (and both color schemes).

**Comment by wrex** [comment]
Sorry. One more commit (again only affects rexs-overrides).

I wanted to document my design goals with the colors. 

As usual, writing things down helped me find a few more minor issues.

**Review by Sepitus-exe — CHANGES_REQUESTED** [review]
I left in a few minor notes, that I would like your input on before merging with the main branch, but overall, very impressive.

**Comment by Sepitus-exe** [comment]
Also I like the pallete you went for, its quite a bit darker and has a richer saturation. Great example of what other looks can be achieved with the framework in place.

> The slider in ultimate timeline still uses the system (OSX) blue slider color, but sliders are a notorious PITA to style.

That slider is the bane of me. I tried my best when I was working on the script theme and eventually got the slider to look semi-passable, but when I was double-checking for compatibility, it was all over the place. Some changes translated to all browsers, but most didn't. Ultimately, I couldn't figure out a solution that would work,so I settled on the default looks.

File: userscript-styles/WKED-self-study.css
Comment (wrex) [reply userscript-styles/WKED-self-study.css:74]: > Is the `!important` necessary? It seems to work just fine without it.

So weird. I need to look at this again. I really hate over-specifying things with `!important` so I'm sure I had a reason (IIRC the original script had a `!important` somewhere that I needed to match).

**Comment by wrex** [comment]
All excellent catches! Thanks for the thorough review!

I'll take care of all of these today.

There's a good article on css-tricks.com on styling range sliders, but it's gross code and I'm not sure if things have improved much since it was written. Not worth it IMO.

I love that we have two wildly different color themes to publish.

File: README.md
Comment (wrex) [reply README.md:222]: Agreed. Changed "the authors" to "we" throughout.

Feel free to make any other edits to the README you'd like.

File: userscript-styles/WKED-self-study.css
Comment (wrex) [reply userscript-styles/WKED-self-study.css:74]: Hmm..

On my system, the `!important` is necessary. Without it, it reverts to the `#8c8` version of green from the self-study quiz styling instead of `--ED-correct`. Same thing for `#f03` and `--ED-incorrect`.

Not sure why it seems to work fine without it for you.

**Comment by wrex** [comment]
Okay, I've addressed everything requested except the `!important` (which I think is required).

Looking forward to getting this into main!

File: userscript-styles/WKED-self-study.css
Comment (Sepitus-exe) [reply userscript-styles/WKED-self-study.css:74]: alright, I will be keeping the `!important` in.

However I will b eremoving the ; from inside the brackets at the line 74. It looks to be a typo that is breking the style on my side.
```suggestion
  color: var(--ED-text-light, #eeeeee) !important;
  background-color: var(--ED-correct, #58896f) !important;
}

#wkof_ds #ss_quiz[data-result="incorrect"] .answer input {
  color: var(--ED-text-light, #eeeeee) !important;
  background-color: var(--ED-incorrect, #9c4644) !important;
```

**Review by Sepitus-exe — APPROVED** [review]
All looks to be in order

**Comment by wrex** [comment]
Woohoo! We're getting close to a release candidate.

What timezone are you in by the way? I'm guessing either an early riser on the east coast of the US or Europe/UK.

I'm based in California (Silly Con valley).

**Comment by Sepitus-exe** [comment]
Quite an educated guess, but I'm in CET, UTC +1:00. As a freshman at a university and with the end of the semester looming over my head, my sleep schedule got messed up with all-nighter after all-nighter. At this point, I wouldn't even describe my sleep situation as scheduled, it's more of an opportunistic thing really.

**Comment by wrex** [comment]
Ah. University. I remember that sleep "schedule".

Okay, based on your perfect English, I'll guess either Swedish, Norwegian, or Serbian/Croation. Less likely, Dutch or German. I don't detect any Germanic notes, nor any latin notes in your English.

**Comment by Sepitus-exe** [comment]
I'm Czech, so my native language is of a slavic descent, which has a nice degree of separation from english as to not interfere with one another. But I did dabble for some six odd years with spanish, never getting the hang of it.

**Comment by wrex** [comment]
Argh! Didn't think about Czech Republic. 😃 

Not a part of the world I frequent, unfortunately. My best friend is half Croatian -- really want to visit together with him someday.

**Pull request merged**
